src/main/java/hudson/plugins/ircbot/v2/IRCColorizer.java

```diff
@@ -66,5 +284,37 @@ private static String colorForBuildResult(String line) {
         }
         return line;
     }
-
+    
+    // write data into a file
+    private static void writeFile(String fileName) {
+        ObjectOutputStream oos = null;
+    	try {
+    		FileOutputStream fout = new FileOutputStream("./"+fileName);
+    		oos = new ObjectOutputStream(fout);
+    		oos.writeObject(userPattern);
+    		oos.close();
```

Comment:
please use the try-finally pattern to close the output stream!
Otherwise it might be left open in case of exceptions